resin/models/supervisor.py

```diff
@@ -0,0 +1,290 @@
+    # _on_device = False, if RESIN_SUPERVISOR_ADDRESS and RESIN_SUPERVISOR_API_KEY env vars are not avaiable => can only communicate with Resin API endpoints.
+    if RESIN_SUPERVISOR_ADDRESS is None or RESIN_SUPERVISOR_API_KEY is None:
+        _on_device = False
+    else:
+        _on_device = True
+
+    def __init__(self):
+        self.base_request = BaseRequest()
+        self.settings = Settings()
+
+    def _check_args(self, device_id, app_id):
+        if device_id is None:
+            exceptions.MissingOption('device_id')
+
+        if app_id is None:
+            exceptions.MissingOption('app_id')
+
+    def _do_command(self, endpoint, req_data={}, device_id=None, app_id=None, method=None):
+        if not self._on_device:
+            self._check_args(device_id, app_id)
+
+            if req_data:
+                data = {
+                    'deviceId': device_id,
+                    'appId': app_id,
+                    'data': req_data
+                }
+            else:
+                data = {
+                    'deviceId': device_id,
+                    'appId': app_id
+                }
+
+            return self.base_request.request(
+                'supervisor/{0}'.format(endpoint),
+                'POST',
+                data=data,
+                endpoint=self.settings.get('api_endpoint'),
+                login=True
+            )
+        else:
+            return self.base_request.request(
+                endpoint,
+                method,
+                data=req_data,
+                endpoint=self.RESIN_SUPERVISOR_ADDRESS,
+                api_key=self.RESIN_SUPERVISOR_API_KEY
+            )
+
+    def force_api_endpoint(self, endpoint):
+        """
+        Force all API commands to a specific endpoint.
+
+        Args:
+            endpoint (bool): True if selecting the API on device. False if selecting the Resin API proxy endpoint.
+
+        Raises:
+            InvalidOption: if endpoint value is not bool.
+
+        """
+
+        try:
+            self._on_device = bool(endpoint)
+        except ValueError:
+            exceptions.InvalidOption('endpoint')
+
+    def ping(self, device_id=None, app_id=None):
+        """
+        Check that the supervisor is alive and well.
+        No need to set device_id and app_id if command is sent to the API on device.
+
+        Args:
+            device_id (Optional[str]): device id.
+            app_id (Optional[str]): application id.
+
+        Returns:
+            str: `OK` signals that the supervisor is alive and well.
+
+        Raises:
+            InvalidOption: if the endpoint is Resin API proxy endpoint and device_id or app_id is not set.
+
+        """
+
+        if not self._on_device:
+            self._check_args(device_id, app_id)
+
+            data = {
+                'deviceId': device_id,
+                'appId': app_id,
+                'method': 'GET'
+            }
+
+            return self.base_request.request(
```

Comment:
Nope, I find this one a bit odd (different endpoint, passing `method` in request data...) in comparison with others. So I want to leave it alone while wrapping others with _do_command